src/main/scala/com/gu/newsletterlistcleanse/braze/BrazeClient.scala

```diff
@@ -1,55 +1,102 @@
 package com.gu.newsletterlistcleanse.braze
 
+import io.circe.Decoder
 import org.slf4j.{Logger, LoggerFactory}
-import scalaj.http._
+import sttp.client._
+import sttp.client.asynchttpclient.future.AsyncHttpClientFutureBackend
+
+import scala.concurrent.duration._
 import io.circe.parser.decode
 import io.circe.syntax._
 
+import scala.concurrent.Future
+import scala.concurrent.ExecutionContext.Implicits.global
+
+
 object BrazeClient {
 
-  private val timeout = 5000
+  private val timeout = 5000.seconds
+  implicit val sttpBackend = AsyncHttpClientFutureBackend(
+    options=SttpBackendOptions.connectionTimeout(timeout)
+  )
 
   val logger: Logger = LoggerFactory.getLogger(this.getClass)
   val brazeEndpoint = "https://rest.fra-01.braze.eu"
 
-  private def withClientLogging[A](info: => String)(block: => Either[BrazeError,A]): Either[BrazeError, A] = {
-    val result = block
+  private def withClientLogging[A](info: => String)(block: => Future[Either[BrazeError,A]]): Future[Either[BrazeError, A]] = {
+    val results = block
 
-    result.foreach { successResult =>
-      logger.info(s"BrazeClient success $info $successResult")
-    }
+    for {
+      result <- results
+      } {
+        result.foreach { successResult =>
+          logger.info(s"BrazeClient success: $info $successResult")
+        }
 
-    result.left.foreach { errorResult =>
-      logger.error(s"BrazeClient failure $info $errorResult")
-    }
+        result.left.foreach { errorResult =>
+          logger.error(s"BrazeClient failure: $info $errorResult")
+        }
+      }
 
-    result
+    results
   }
 
-  private def parseValidateResponse(response: HttpResponse[String]): Either[BrazeError, BrazeResponse] = {
-    decode[BrazeResponse](response.body) match {
-      case Right(parsedResponse) if response.is2xx  =>
-        Right(parsedResponse)
-      case Left(e) if response.is2xx =>
-        logger.error("failure to parse Braze response", e)
-        Left(BrazeError(response))
-      case _ =>
-        Left(BrazeError(response))
+  private def parseValidateResponse[T: Decoder](response: Future[Response[Either[String, String]]]): Future[Either[BrazeError, T]] = {
```

Comment:
It's usually nicer to receive parameters that aren't futures. It's not always possible (for instance `withClientLogging` it makes sense there), but I believe in this case it can be avoided. So within the context of that function, everything is synchronous. However, when calling it you make use of the asynchronous context like so

```scala
response.map(parseValidateResponse[SimpleBrazeResponse])
```

Reply:
That's a really nice design pattern and much more understandable than passing `Future`s around everywhere. Thanks!